cpuwindow.cpp

```diff
@@ -220,4 +220,130 @@ void CPUwindow::connectBackend()
+                break;
+            case 3:
+                this->ui->r3->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r3->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 4:
+                this->ui->r4->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r4->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 5:
+                this->ui->r5->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r5->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 6:
+                this->ui->r6->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r6->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 7:
+                this->ui->r7->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r7->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 8:
+                this->ui->r8->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r8->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 9:
+                this->ui->r9->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r9->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 10:
+                this->ui->r10->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r10->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 11:
+                this->ui->r11->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r11->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 12:
+                this->ui->r12->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r12->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 13:
+                this->ui->r13->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r13->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 14:
+                this->ui->r14->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r14->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            case 15:
+                this->ui->r15->setStyleSheet("color: rgb(239, 41, 41);");
+                this->ui->r15->setText("0x" + QString::number(value, 16).toUpper());
+                break;
+            default:
+                break;
+            }
+        }
+        else {
+            this->ui->GRlineR->setStyleSheet("color: rgb(0, 0, 0);");
+            this->ui->RGview->setStyleSheet("color: rgb(0, 0, 0);");
+
+            this->ui->r0->setStyleSheet("color: rgb(0, 0, 0);");
```

Comment:
Set the values of registers to 0

Reply:
This is quite hard. We want to maintain registers to their values when deactivating lines (signals). Reset should be done only when reassembling. Emitting signal on constructor has no effect, so here a much more complex approach is needed. I'd postpone it for later, but as you consider.

Reply:
Done